irods/test/data_obj_test.py

```diff
@@ -1691,6 +1695,31 @@ def replica_number_from_resource_name(data_path, resc):
             if dir1: self.sess.resources.get(uniq1).remove()
             if dir2: self.sess.resources.get(uniq2).remove()
 
+    def _auto_close_test(self, name, content):
+        d = None
+        try:
+            d = self.sess.data_objects.get(name)
+            self.assertEqual(int(d.replicas[0].status), 1)
+            self.assertEqual(d.open('r').read().decode(), content)
+        finally:
+            if d: d.unlink(force = True)
+
+
+    @staticmethod
+    def _python(): return 'python'+str(sys.version_info[0])
+
+    def test_data_objects_auto_close_on_process_exit__issue_456(self):
+        program = os.path.join(test_modules.__path__[0],'close_data_objects.py')
```

Comment:
pro: it can be run as its own executable, which is by design... we're testing what happens when the process exits.
con:  I couldn't get that to happen if the code were in the same module.  I tried one other way, with multiprocessing.Process, but the forked process calls `_exit` ie doesn't do clean-up

Reply:
Understood. If you've added words explaining the **why**, then that is good enough for me.